src/spec-node/singleContainer.ts

```diff
@@ -159,14 +164,21 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 
 	const args: string[] = [];
 	if (buildParams.buildKitVersion) {
-		args.push('buildx', 'build',
-			'--load', // (short for --output=docker, i.e. load into normal 'docker images' collection)
-			'--build-arg', 'BUILDKIT_INLINE_CACHE=1', // ensure cache manifest is included in the image
-		);
+		args.push('buildx', 'build');
+		if (buildParams.buildxPlatform) {
+			args.push('--platform', buildParams.buildxPlatform);
+		}
+		if (buildParams.buildxPush) {
+			args.push('--push');
+		} else {
+			args.push('--load'); // (short for --output=docker, i.e. load into normal 'docker images' collection)
+		}
+		args.push('--build-arg', 'BUILDKIT_INLINE_CACHE=1');
 	} else {
 		args.push('build');
```

Comment:
Can we fail here (when BuildKit is not available) when any of the new options are set? That will make it easier for the user to understand what is happening.